File: frontend/lib/data/models/property_model.dart
class PropertyModel {
  final String id;
  final String projectName;
  final List<String> images;
  final String address;
  final double latitude;
  final double longitude;
  final int price;
  final String type;
  final String description;
  final String createdAt;
  final String updatedAt;

  PropertyModel({
    required this.id,
    required this.projectName,
    required this.images,
    required this.address,
    required this.latitude,
    required this.longitude,
    required this.price,
    required this.type,
    required this.description,
    required this.createdAt,
    required this.updatedAt,
  });

  factory PropertyModel.fromMap(Map<String, dynamic> map) {
    return PropertyModel(
      id: map['_id'] ?? '',
      projectName: map['projectName'] ?? '',
      images: List<String>.from(map['images'] ?? []),
      address: map['address'] ?? '',
         latitude: (map['latitude'] as num?)?.toDouble() ?? 0.0, // ✅ Null check with fallback value
      longitude: (map['longitude'] as num?)?.toDouble() ?? 0.0,
      price: map['price'] ?? '',
      type: map['type'] ?? '',
      description: map['description'] ?? '',
      createdAt: map['createdAt'] ?? '',
      updatedAt: map['updatedAt'] ?? '',
    );
  }
}

File: Backend/Models/property.model.js
import mongoose from "mongoose";

const propertySchema = new mongoose.Schema(
    {

        userId: {
            type: mongoose.Schema.Types.ObjectId,
            ref: "User",
            required: true,
        },
        projectName : {
            type : String,
            required : true,
            trim : true,
        },
        images: [
            {
              type: String,
              required: true,
            }
          ],
          address: {
            type: String,
            required: true,
          },
          area: {
            type: Number,
            required: true,
          },
          availableArea: {
            type: Number,
            required: true,
          },
          latitude: {
            type: Number,
            required: true,
          },
          longitude: {
            type: Number,
            required: true,
          },
          price: {
            type: String,
            required: true,
          },
          type: {
            type: String,
            enum: ["Residential", "Commercial", "Holiday Homes","Plot"],
            required: true,
          },
          description: {
            type: String,
            required: true,
          },
        }, {
          timestamps: true
        }
);

export const Property = mongoose.model("Property",propertySchema);